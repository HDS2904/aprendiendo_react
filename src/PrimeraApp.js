import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// Funsional components
const PrimeraApp = ( { saludo, subtitulo } ) => {
    
    return (
        <>
            <h1>PropTypes</h1>
            <h3>{ saludo }</h3>
            <p>{ subtitulo }</p>
        </>
    );
}

// Código para obligar el envio de parametros "required"
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

// Definición de contedido de los props por defecto
PrimeraApp.defaultProps = {
    subtitulo: 'Un subtitulo por defecto'
}

export default PrimeraApp;