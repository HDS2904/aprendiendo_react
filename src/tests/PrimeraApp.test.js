import React from 'react'
import { shallow } from 'enzyme';
// import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp"

describe('Pruebas en <PrimeraApp/>', () => {
    // test('Prueba que retorna un string de saludo', () => {
    //     const saludo = 'Hola soy Jonathan'
    //     const { getByText } = render( <PrimeraApp saludo={saludo} /> )
    //     expect( getByText(saludo) ).toBeInTheDocument( )
    // })
    
    test('Prueba que retorna <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola soy Jonathan'
        const wrapper = shallow( <PrimeraApp saludo={saludo} /> )
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Prueba con subtitulo del snapshot', () => {
        const greet = 'Hola soy Jonathan'
        const subtitle = 'Un subtitulo por defecto'
        const wrapper = shallow( <PrimeraApp saludo={greet} subtitulo={subtitle} /> )
        const textTest = wrapper.find('p').text();
        expect( textTest ).toBe( subtitle )
        
    })

})
