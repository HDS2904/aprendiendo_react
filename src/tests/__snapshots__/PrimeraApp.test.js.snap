// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Prueba que retorna <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    PropTypes
  </h1>
  <h3>
    Hola soy Jonathan
  </h3>
  <p>
    Un subtitulo por defecto
  </p>
</Fragment>
`;
